models/imagick.go

```diff
@@ -131,6 +141,39 @@ func processImage(tempDir string, inFile string, args *ProcessArgs) (string, err
 	return outFileWithFormat, err
 }
 
+func postProcessImage(tempDir string, inFile string, args *ProcessArgs) (string, error) {
+	// If it originally "mp4" was requested even if before processing
+	// changed it to "gif"
+	grohl.Log(grohl.Data{"args": args})
+	if args.RequestFormat == "mp4" && args.Format == "gif" {
+		outFile := filepath.Join(tempDir, "video.mp4")
+		cmdArgs := []string{"-f", "gif", "-i", inFile, outFile}
+
+		grohl.Log(grohl.Data{
+			"processor": "ffmpeg",
+			"args":      cmdArgs,
+		})
+
+		cmd := exec.Command("ffmpeg", cmdArgs...)
+		var outErr bytes.Buffer
+		cmd.Stdout, cmd.Stderr = &outErr, &outErr
+		err := runWithTimeout(cmd, normalTimeout)
+		if err != nil {
+			grohl.Log(grohl.Data{
+				"processor": "ffmpeg",
+				"step":      "post-process-mp4",
+				"failure":   err,
+				"args":      cmdArgs,
+				"output":    string(outErr.Bytes()),
+			})
+		}
+
+		return outFile, err
+	}
+
+	return inFile, nil
+}
```

Comment:
Wish there was a way to do this with imagemagick, but this looks like a good solution.